lib/Conversion/TritonGPUToLLVM/AssertOpToLLVM.cpp

```diff
@@ -34,19 +34,30 @@ struct AssertOpConversion : public ConvertOpToLLVMPattern<triton::AssertOp> {
         return failure();
       }
     }
-    llAssert(op, condition, adaptor.getMessage(), adaptor.getFile(),
-             adaptor.getFunc(), adaptor.getLine(), rewriter);
+    llAssert(op, condition, adaptor.getMessage(), rewriter);
     rewriter.eraseOp(op);
     return success();
   }
   // op: the op at which the assert is inserted. Unlike printf, we need to
   // know about the op to split the block.
   void llAssert(Operation *op, Value condition, StringRef message,
-                StringRef file, StringRef func, int line,
                 ConversionPatternRewriter &rewriter) const {
     ConversionPatternRewriter::InsertionGuard guard(rewriter);
+
     auto ctx = rewriter.getContext();
     auto loc = op->getLoc();
+
+    StringRef file = "unknown";
+    StringRef func = "unknown";
+    int line = 0;
+    int col = 0;
+
+    if (auto fileLineColLoc = dyn_cast<FileLineColLoc>(loc)) {
```

Comment:
In what cases it's not a `FileLineColLoc`?

Reply:
The cast fails when `TRITON_DISABLE_LINE_INFO` is set to 1